decorator.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name.slice(0, 10) : @nameable.correct_name
+  end
+end
```

Comment:
Please move these two classes to their separate files. This way the project files are managed and the code is clean.

Reply:
Thank you for your review! I have moved each decorator to its own file to better manage the project.